proto/fulfillment/v1/host_class_type.proto

```diff
@@ -0,0 +1,41 @@
+//
+// Copyright (c) 2025 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+//
+
+syntax = "proto3";
+
+package fulfillment.v1;
+
+import "shared/v1/metadata_type.proto";
+
+// Describes a set of hosts that share characteristics.
+//
+// For example there could be a host class `acme_1tb` to describe the set of hosts manifactured by ACME and with 1 TiB
+// of RAM, and another `ibm_mi300x` to describe the set of hosts manufactured IBM and with a MI300X GPU.
+//
+// This is similar to the _instance type_ concept used by many cloud providers.
+//
+// The detailed chracteristics of the host (CPU, memory, GPU, etc) will be in the `description` field.
+message HostClass {
```

Comment:
Does this have any direct relation to the ClusterNodeSet? How would this API get used? Is this a way to list the classes I can choose from, and if so, what value from here do I put into the ClusterNodeSet `class` field?

Reply:
The relation is that the key of the map of node sets (or the `name` of the node set, if we use an array instead of a map) will be the identifier of one of these objects. So if the user sees a node set with key (or name) "xyz23" she can fetch the details of host class "xyz23" and read the description field to understand what that means.

The user can't set the  `class` field, it is read only.

In the future we may want to use this host class concept for bare metal provisioning, but for now it is just informative.